### DISCUSSION TITLE
"Generating new OAuth keys" after each container restart

### DISCUSSION THREAD

**Discussion by Fmstrat** [firefly-iii/firefly-iii#8635] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

https://github.com/firefly-iii/firefly-iii/issues/7630 has returned. All details are specific in that ticket. Every startup creates two client IDs that do not show up in the UI:
```
|budget-firefly | 2024-03-02T20:52:30.828863765Z Encryption keys already exist. Use the --force option to overwrite them.
|budget-firefly | 2024-03-02T20:52:30.841587663Z Personal access client created successfully.
|budget-firefly | 2024-03-02T20:52:30.841608966Z Client ID: 3
|budget-firefly | 2024-03-02T20:52:30.841635021Z Client secret: ...
|budget-firefly | 2024-03-02T20:52:30.844434447Z Password grant client created successfully.
|budget-firefly | 2024-03-02T20:52:30.844458172Z Client ID: 4
|budget-firefly | 2024-03-02T20:52:30.844493528Z Client secret: ...
|budget-firefly | 2024-03-02T20:52:31.005217029Z   [i] Updated version.
|budget-firefly | 2024-03-02T20:52:31.507934326Z   [✓] No security alerts for version 6.1.9
```

### Debug information

Debug information generated at 2024-03-02 15:55:36 for Firefly III version **v6.1.9**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.9 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.2 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>957, base <span>#</span>79</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>America/New_York + America/New_York</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td>2024-03-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-03-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0</td></tr></tbody></table>

### Steps to reproduce

1. Fresh install using docker
2. It generates client ID 1 and 2
3. Restart container
4. It generates client ID 3 and 4
5. So on...

### Additional info
- Unsure if related, but when trying to use the importer, and using client ID `8`, which is the client I created in the UI, I get an `invalid_client`, `Client authorization failed`

**Comment by JC5** [comment]
I can't replicate this at all. It seems to work just fine. What's your host OS?

**Comment by Fmstrat** [comment]
I will assemble a docker compose that can be run from scratch to reproduce, if that helps?

**Comment by JC5** [comment]
Yes please.

**Comment by Fmstrat** [comment]
Here's a fully contained `docker-compose` that reproduces the issue. If you start it, bring it down, and bring it back up, you'll see it creates new client IDs each time. Hope this helps!

[docker-compose.txt](https://github.com/firefly-iii/firefly-iii/files/14497680/docker-compose.txt)

**Comment by JC5** [comment]
Ah yes I see it. This is normal, unfortunately. 

The client_id I can't confirm. Did you create a non-confidential client?

**Comment by Fmstrat** [comment]
I'm not sure I understand what you mean by non-confidential, sorry.

**Reply by JC5** [reply]
https://docs.firefly-iii.org/how-to/firefly-iii/features/api/#confidential

**Reply by Fmstrat** [reply]
Ahhh that could be related. I couldn't get the OAuth to work because no matter what I did the redirect went to localhost, and since I'm running on a server in docker, the importer wasn't on localhost. Since it's behind a secure network, I used a PAT. But no, I didn't make it confidential.
